question25/main.js

```diff
@@ -0,0 +1,115 @@
+const modalClose = document.getElementById('js-close');
+const mask = document.getElementById('js-mask');
+const checkbox = document.getElementById('js-checkbox');
+const submitButton = document.getElementById('js-submit-button');
+const modalContainer = document.getElementById("js-modal-container");
+const nameInputArea = document.getElementById('name');
+const mailInputArea = document.getElementById('mail');
+const passwordInputArea = document.getElementById('password');
+
+
+textLinkToTerms.addEventListener('click',()=>{
+    modal.classList.remove('hidden');
+    mask.classList.remove('hidden');
+});
+modalClose.addEventListener('click',()=>{
+    modal.classList.add('hidden');
+    mask.classList.add('hidden');
+});
+mask.addEventListener('click',()=>{
+    modalClose.click();
+});
+
+const options = {
+  root: document.getElementById('js-modal'),
+  threshold: 1
+};
+
+const checkWhenIntersect = ([entry]) => {
+  checkbox.checked = entry.isIntersecting;
+  checkbox.disabled = !entry.isIntersecting;
+  entry.isIntersecting && checkAllValidity();
+};
+
+const observer = new IntersectionObserver(checkWhenIntersect, options);
+observer.observe(modalContainer.lastElementChild);
+
+
+
+const isValidStatus = {
+  name: false,
+  mail: false,
+  password: false,
+}
+
+const validationInfo = {
+  name: {
+    maxNameLength: 16,
+    minNameLength: 1,
+    validation: (value) => value.length > validationInfo.name.minNameLength && value.length < validationInfo.name.maxNameLength,
+    errorMessage: '※ユーザー名は1文字以上15文字以下にしてください。',
+  },
+  mail: {
+    //Reference: https://1-notes.com/javascript-determine-if-it-is-an-email-address/ 
+    validation : (value) => /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(value),
+    errorMessage: '※メールアドレスの形式になっていません。' 
+  },
+  password: {
+    validation : (value) => /(?=.*[a-zA-Z])(?=.*[0-9])[a-zA-z0-9]{8,}/.test(value),
+    errorMessage: '※8文字以上の大小の英数字を交ぜたものにしてください。'  
+  }
+}
+
+const checkFormValidation = (isValid, targetForm) => {
+  if (isValid) {
+    targetForm.classList.add('valid');
+    targetForm.classList.remove('invalid');
+    targetForm.nextElementSibling.textContent = ''; 
+    isValidStatus[targetForm.id] = true;
+  } else {
+    targetForm.classList.add('invalid');
+    targetForm.classList.remove('valid');
+    targetForm.nextElementSibling.textContent = validationInfo[targetForm.id].errorMessage;
+    isValidStatus[targetForm.id] = false; 
+  }
+}
+
+const checkInputValue = (e) => {
+  const targetForm = e.target;
+  const value = targetForm.value.trim();
+  const result = validationInfo[targetForm.id].validation(value);
+  checkFormValidation(result,targetForm);
+  checkEmptyCharacter(targetForm);  
+  checkAllValidity();
+} 
+
+const checkAllValidity = () => {
+  const isValid = Object.values(isValidStatus);
+  const allValidStatus = isValid.every((result) => { return result });
```

Comment:
IMO
Since the return value is a boolean, is** is better.
Also, return can be omitted.
```suggestion
  const isAllValid  = isValid.every((result) => result);
```

Reply:
Thank you for your review! I still have a lot to learn. I fied so please confirm.
fixed commit : 
https://github.com/csakyo/homework/pull/39/commits/ee204455428111873b504ca018311d588602f8a2
https://github.com/csakyo/homework/pull/39/commits/8dc533e7f33d03f4c9a08ad50ecb1c8ce6625c50